src/pre_defined_classes/programs_with_changing_behaviour/hard/q2/Question2Test.java

```diff
@@ -0,0 +1,25 @@
+package pre_defined_classes.programs_with_changing_behaviour.hard.q2;
+
+import global.BaseRandomTest;
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.RandomInteger;
+import global.variables.clauses.StringLiteral;
+
+public class Question2Test extends BaseRandomTest {
+
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new RandomInteger(10000000, 100000000),
+                new StringLiteral(" has been decrypted using "),
+                new RandomInteger(1, 10),
+                new StringLiteral(" into "),
+                new IntegerLiteral(1000000, 100000000)
+        };
+    }
+
+    public void runMain() {
+        Question2.main(new String[0]);
+    }
+}
```

Comment:
I think this distractor:
int encrypt = crypt / gen.nextInt( 10 );
could still cause this test to pass, maybe still include a test that checks the math?

Reply:
`gen.nextInt( 10 );` could return `0` so this solution would fail if we ran it enough times (which BaseRandomTest does) just because dividing by 0 is an error.

Reply:
Oh, so because there's one random in it, it runs the whole test a bunch of times, not just that one line?

Reply:
Yeee anything that `extends BaseRandomTest` gets run somewhere around 500 times

Reply:
Should I warp it in a try-catch then?

Reply:
I inserted a try-catch block, so that should deal with that issue.